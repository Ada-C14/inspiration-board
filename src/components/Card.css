.card {
  background-color: #F4FF81;

  padding: 1em 0;
  margin: 0.5rem;

  min-height: 250px;
  min-width: 225px;
  flex-basis: 18%;

  border-radius: 5px;

  display: grid;
  grid-template-columns: 1fr 5fr 1fr;
  grid-template-rows: 3fr 1fr;
  align-items: center;
}

.card__content {
  grid-column-start: 2;

  display: flex;
  flex-direction: column;
  justify-content: space-around;

  text-align: center;
  font-weight: 100;
  overflow: hidden;
}

.card__content-text {
  text-overflow: ellipsis;
  overflow: hidden;

  font-family: 'Permanent Marker', Helvetica, sans-serif;
}

.card__content-emoji {
  font-size: 3rem;
}

.card__content-text, .card__content-emoji {
  padding: 0;
  margin: 0;
}

.card__delete-button {
  align-self: start;
  font-family: 'Permanent Marker', Helvetica, sans-serif;
}

.card__delete {
  grid-row-start: 2;
  grid-column-start: 2;

  display: flex;
  justify-content: space-around;
  align-items: top;

}