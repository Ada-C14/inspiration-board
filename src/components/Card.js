import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';


const Card = (props) => {
  
  return (
    <div className="card" id={props.id} >
      {props.text}
      {emoji.getUnicode(`${props.emoji}`)}
      <button onClick={props.onDelete} id={props.id}>
        Delete
      </button>
    </div>
  )
}

Card.propTypes = {
  text:PropTypes.string,
  emoji:PropTypes.string
};

export default Card;
