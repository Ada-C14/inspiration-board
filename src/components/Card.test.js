import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import { render, cleanup } from '@testing-library/react'


describe('Card', () => {
  test('that it matches the existing snapshot of Card', () => {
    // Arrange-Act
    const { asFragment } = render(
      <Card
      text="new card for test"
      onDelete={() => { }}
      />
      // <Card key={card.card.id} id={card.card.id} text={card.card.text} emoji={card.card.emoji} onDelete={props.deleteCard} />
    );

    // Assert
    expect(asFragment()).toMatchSnapshot();
    cleanup();
  });
});
