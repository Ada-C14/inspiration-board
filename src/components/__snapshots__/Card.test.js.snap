// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card that it matches the existing snapshot of Card 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    new card for test
    <button>
      Delete
    </button>
  </div>
</DocumentFragment>
`;
